forecasting/src/autogluon/forecasting/dataset/ts_dataframe.py

```diff
@@ -0,0 +1,226 @@
+            1. Time-series data in TimeSeriesListDataset format.
+            2. Time-series data pd.DataFrame format with multi-index on item and timestamp.
+            An example can be found using example_dataframe() function.
+    """
+
+    def __init__(self, data: Any, *args, **kwargs):
+        if isinstance(data, TimeSeriesListDataset):
+            data = self.from_list_dataset(data)
+        self._validate_data_frame(data)
+        super().__init__(data=data, *args, **kwargs)
+
+    @classmethod
+    def example_dataframe(cls):
+        """An example TimeSeriesDataFrame.
+
+        Returns
+        -------
+        ts_df : TimeSeriesDataFrame
+            It returns an example TimeSeriesDataFrame as:
+                                    target
+            item_id timestamp
+            0       2019-01-01       0
+                    2019-01-02       1
+                    2019-01-03       2
+            1       2019-01-01       3
+                    2019-01-02       4
+                    2019-01-03       5
+            2       2019-01-01       6
+                    2019-01-02       7
+                    2019-01-03       8
+        """
+
+        target = list(range(9))
+        datetime_index = tuple(
+            pd.date_range(pd.Timestamp("01-01-2019"), periods=3, freq="D")
+        )
+        item_ids = (0, 1, 2)
+        multi_index = pd.MultiIndex.from_product(
+            [item_ids, datetime_index], names=["item_id", "timestamp"]
+        )
+        ts_df = TimeSeriesDataFrame(
+            pd.Series(target, name="target", index=multi_index).to_frame()
+        )
+        return ts_df
+
+    @classmethod
+    def _validate_data_frame(cls, data: pd.DataFrame):
+        """Validate a pd.DataFrame can be converted to TimeSeriesDataFrame
+
+        Parameters:
+        -----------
+        data : pd.DataFrame
+            a data frame in pd.DataFrame format.
+        """
+
+        if not isinstance(data, pd.DataFrame):
+            raise ValueError(f"data must be a pd.DataFrame, get {type(data)}")
+        if not isinstance(data.index, pd.MultiIndex):
+            raise ValueError(f"data must have pd.MultiIndex, get {type(data.index)}")
+        if "target" not in data.columns:
+            raise ValueError(f"data must have a column called target")
+        if not data.index.dtypes.array[0] == np.dtype(np.int64):
+            raise ValueError(
+                f"for item_id, the only NumPy dtype allowed is ‘np.int64’."
+            )
+        if not data.index.dtypes.array[1] == np.dtype("datetime64[ns]"):
+            raise ValueError(
+                f"for timestamp, the only NumPy dtype allowed is ‘datetime64[ns]’."
+            )
+
+    @classmethod
+    def _validate_list_dataset(cls, list_dataset):
+        if not isinstance(list_dataset, TimeSeriesListDataset):
+            raise ValueError(
+                f"list_dataset must be a TimeSeriesListDataset, get {type(list_dataset)}"
```

Comment:
nitpick: maybe "got" or "not"?

Reply:
similarly for below

Reply:
Noted, will change in the next revision.